models/Asset.php

```diff
@@ -164,7 +167,12 @@ protected function getBlockedVars(): array
 
     public static function getTypes(): array
     {
-        return self::$types;
+        $assetsConfig = Config::getSystemConfiguration('assets');
+        if (isset($assetsConfig['type_definitions']['map']) && is_array($assetsConfig['type_definitions']['map'])) {
```

Comment:
shouldn't the AssetTypeLoader gather that? The Configuration is only one implementation of it.

Reply:
So you would call the AssetTypeLoader here to return the types right?